test/setup.ts

```diff
@@ -111,196 +107,37 @@ export async function setup() {
   let bridgeOracle: DSValue;
   let host: DomainHost;
   let guest: SnDomainGuest;
+  let mockStarknetMessaging: Address;
 
-  // If setupconfig is not an empty object, we skip deployments and get directly the contracts from this object
-  if (Object.keys(setupConfig).length !== 0) {
-    // Get teleport contracts
-    console.log("getTeleport");
-    teleport = await getTeleport(
-      setupConfig.l1join,
-      setupConfig.l1router,
-      setupConfig.l1oracleauth
-    );
-
-    // Get fees contracts
-    console.log("getLinearFee");
-    fees = await getLinearFee(setupConfig.l1fees);
-
-    console.log("getSnDss");
-    snDss = await getSnDss(
-      setupConfig.l2vat,
-      setupConfig.l2jug,
-      setupConfig.l2cure,
-      snCfg.dai,
-      setupConfig.l2daiJoin,
-      setupConfig.l2end,
-      setupConfig.l2spotter,
-      setupConfig.l2pot
-    );
-
-    console.log("getSnToken");
-    snClaimToken = await getSnToken(setupConfig.claimToken);
-
-    console.log("getSnTeleport");
-    snTeleport = await getSnTeleport(
-      setupConfig.l2join,
-      setupConfig.l2router,
-      setupConfig.l2oracleauth
-    );
-
-    console.log("getSnDomainGuest");
-    guest = await getSnDomainGuest(setupConfig.domainGuest);
-
-    console.log("getSnTeleportConstantFee");
-    snFee = await getSnTeleportConstantFee(setupConfig.l2fees);
-
-    console.log("getDomainHost");
-    host = await getDomainHost(setupConfig.domainHost);
-
-    console.log("getBridgeOracle");
-    bridgeOracle = await getBridgeOracle(setupConfig.bridgeOracle);
-
-    startL1Prank(admin);
-    startSnPrank(snOwner);
-  } else {
-    console.log("deployTeleport");
-    teleport = await deployTeleport(
-      deployer,
-      admin,
-      rootCfg.teleportIlk,
-      snCfg.domain,
-      rootCfg.domain,
-      dss.daiJoin
-    );
-
-    console.log("deployLinearFee");
-    fees = await deployLinearFee(WAD / 10000n, _6_HOURS);
-
-    hostAddress = (await getAddressOfNextDeployedContract()) as Address;
-
-    // deploy on Starknet
-    console.log("deploySnDss");
-    snDss = await deploySnDss(snOwner, snCfg.dai);
-
-    console.log("deploySnToken");
-    snClaimToken = await deploySnToken(snOwner.address);
-    // TODO: snClaimToken rely, deny
-    // claimToken.rely(radmin);
-    // claimToken.deny(address(this));
-
-    console.log("deploySnTeleport");
-    snTeleport = await deploySnTeleport(
-      snOwner,
-      snCfg.teleportIlk,
-      snCfg.domain,
-      rootCfg.domain,
-      snDss.daiJoin
-    );
-
-    console.log("deploySnDomainGuest");
-    guest = await deploySnDomainGuest(
-      snOwner,
-      snDss.daiJoin,
+  // load or deploy
+  if (load) {
```

Comment:
I would slightly reorganize, more or less like this::
```
  const { teleport, snTeleport, snDss, fees, snFee, host, guest } = snapshotExists()
    ? await loadSnaphost(admin, snOwner, snCfg)
    : await doSetup(dss, deployer, admin, snOwner, snCfg, rootCfg);
```
`SetupConfig` should stay a private detail of `loadSnapshot`.
It seems that `dss` initialization can be moved inside of `doSetup`.